csi/deployment.go

```diff
@@ -307,8 +307,7 @@ func NewPluginDeployment(namespace, serviceAccount, nodeDriverRegistrarImage, li
 					Tolerations:        tolerations,
 					NodeSelector:       nodeSelector,
 					PriorityClassName:  priorityClass,
-					HostNetwork:        true,
-					DNSPolicy:          corev1.DNSClusterFirstWithHostNet,
+					HostPID:            true,
```

Comment:
The drawback of reverting to the previous implementation is that it would allow the pod to view all host processes and also the ability to gain control of the host.

What about removing longhorn-csi-plugin from the NetworkPolicy? What are the security risks here?

Reply:
> The drawback of reverting to the previous implementation is that it would allow the pod to view all host processes and also the ability to gain control of the host.

I see. Yeah.

> What about removing longhorn-csi-plugin from the NetworkPolicy? What are the security risks here?

But I don't know how to do this. The network policy is to add more pod into the allow list. If we don't add longhorn-csi-plugin into the policy it will be blocked by default. Any idea?

Reply:
> But I don't know how to do this. The network policy is to add more pod into the allow list. If we don't add longhorn-csi-plugin into the policy it will be blocked. Any idea?

Does CIDR notation work?

Reply:
To whitelist the host IPs? 

I think host IPs are dynamic depending on each cluser setup? 

Also, this mean we allow any host processes to hit the longhorn-manager pods which sounds a bit off to me

WDYT?

Reply:
Yea, true.

Reply:
>And, how do we handle the scenario where both the network policy and storage network for RWX volume are enabled? Could this be a use case for Harvester?

Sorry @c3y1huang, I am confused with that.

It looks like we will get hostPID back, right? If yes, hostPID will break the storage network. (from the comment https://github.com/longhorn/longhorn/issues/8184#issuecomment-2124179154)
Does it mean we need to find another way to support the storage network on the RWX volume?

Reply:
We will get the `hostPID` and the custom `mounter` back. The custom mounter is a wrapper of `nsenter` around the mounter commands to execute in the host namespace, we will only apply this to regular RWX volumes. For storage network enabled RWX volumes, the default mounter will be used, this means the mounting will be done within the CSI plugin pod container. Therefore, this change shouldn't impact storage network support for RWX volumes.

Reply:
Hi @c3y1huang,
That's clear.
- Regular RWX -> use nsenter for custom mounter
- Storage Network RWX -> use normal mounted

Thanks for the explanation for it!